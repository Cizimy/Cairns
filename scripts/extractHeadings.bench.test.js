import { jest, describe, test, expect } from '@jest/globals';
import { extractHeadings } from './generate-prompt.js'; // テスト対象をインポート

describe('extractHeadings Benchmark', () => {
  test('should handle markdown with a large number of headings (snapshot test)', () => {
    const headingCount = 1000;
    let largeMarkdown = '';
    for (let i = 1; i <= headingCount; i++) {
      const level = (i % 6) + 1; // Cycle through levels 1-6
      largeMarkdown += `${'#'.repeat(level)} Heading ${i} {#id-${i}}\nSome text under heading ${i}.\n`;
    }

    // console.time('extractHeadingsLarge'); // 計測開始 (デバッグ用)
    const headings = extractHeadings(largeMarkdown);
    // console.timeEnd('extractHeadingsLarge'); // 計測終了 (デバッグ用)

    // 結果の件数を確認
    expect(headings).toHaveLength(headingCount);

    // 結果の内容をスナップショットで確認 (実行時間ではなく、出力の一貫性を担保)
    // スナップショットには最初の5件と最後の5件のみを含める（ファイルサイズを抑えるため）
    const snapshotData = {
        firstFive: headings.slice(0, 5),
        lastFive: headings.slice(-5),
        totalCount: headings.length,
    };
    expect(snapshotData).toMatchSnapshot();
  });
});