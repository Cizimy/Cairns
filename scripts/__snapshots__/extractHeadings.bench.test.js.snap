// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractHeadings Benchmark should handle markdown with a large number of headings (snapshot test) 1`] = `
{
  "firstFive": [
    {
      "level": 2,
      "title": "Heading 1",
    },
    {
      "level": 3,
      "title": "Heading 2",
    },
    {
      "level": 4,
      "title": "Heading 3",
    },
    {
      "level": 5,
      "title": "Heading 4",
    },
    {
      "level": 6,
      "title": "Heading 5",
    },
  ],
  "lastFive": [
    {
      "level": 1,
      "title": "Heading 996",
    },
    {
      "level": 2,
      "title": "Heading 997",
    },
    {
      "level": 3,
      "title": "Heading 998",
    },
    {
      "level": 4,
      "title": "Heading 999",
    },
    {
      "level": 5,
      "title": "Heading 1000",
    },
  ],
  "totalCount": 1000,
}
`;
